### PR TITLE
Add API for getblockheader

### LINKED ISSUE #89: getblockheader is not exposed
`getblockheader ` is an important RPC call that is not exposed. At first glance it looks the same as `getblock` and dropping the `tx` array, but there's an important difference:

```
elsa@winter:~$ bitcoin-cli getblockheader 000000000000000000f37fddab6ae59b06d55c9949c4bf35151b7776ff551897
{
  "hash": "000000000000000000f37fddab6ae59b06d55c9949c4bf35151b7776ff551897",
  "confirmations": 100001,
  "height": 329936,
  "version": 2,
  "versionHex": "00000002",
  "merkleroot": "48786f412860f93607fd2b67bfc32d1c905c83da12efd35f5718b38da44b296b",
  "time": 1415947066,
  "mediantime": 1415942411,
  "nonce": 1327336627,
  "bits": "181bc330",
  "difficulty": 39603666252.41841,
  "chainwork": "00000000000000000000000000000000000000000002ba5d1b2f3f765cff293c",
  "previousblockhash": "00000000000000000b954ce608c45c9229c3d8e0f8f710e5663e0bb9091d33e1",
  "nextblockhash": "000000000000000000657614104babe20c0f76066c288c0b849504ea11f7af6e"
}
elsa@winter:~$ bitcoin-cli getblock 000000000000000000f37fddab6ae59b06d55c9949c4bf35151b7776ff551897
error code: -32603
error message:
Block not available (pruned data)
```

### PR DESCRIPTION
**Pull request by dajohi**
GetBlockHeader returns the serialized blockheader.
GetBlockHeaderVerbose returns the data structure with information about
the blockheader.

Fixes #89

### CODE REVIEW

**Comment by FrozenPrincess** [comment]
I checked out this branch and tested it against bitcoin core, works great!

File: chain.go [hunk 1]
```diff
@@ -263,6 +263,92 @@ func (c *Client) GetBlockHash(blockHeight int64) (*chainhash.Hash, error) {
 	return c.GetBlockHashAsync(blockHeight).Receive()
 }
 
+// FutureGetBlockHeaderResult is a future promise to deliver the result of a
+// GetBlockHeaderAsync RPC invocation (or an applicable error).
+type FutureGetBlockHeaderResult chan *response
+
+// Receive waits for the response promised by the future and returns the
+// serialized blockheader requested from the server given its hash.
+func (r FutureGetBlockHeaderResult) Receive() (string, error) {
```
Comment (davecgh) [inline chain.go]: This should return a `*wire.BlockHeader` in my opinion so it's consistent with the other block-relatedAPIs which return concrete types as opposed to the raw JSON types (the hex in this case).

See the `func (r FutureGetBlockResult) Receive() (*btcutil.Block, error) {` for an example and details on how it deserializes.

File: chain.go [hunk 2]
```diff
@@ -263,6 +263,92 @@ func (c *Client) GetBlockHash(blockHeight int64) (*chainhash.Hash, error) {
...
+
+	cmd := btcjson.NewGetBlockHeaderCmd(hash, btcjson.Bool(false))
+	return c.sendCmd(cmd)
+}
+
+// GetBlockHeader returns the serialized blockheader from the server given its hash.
+//
+// See GetBlockHeaderVerbose to retrieve a data structure with information about the
+// block instead.
+func (c *Client) GetBlockHeader(blockHash *chainhash.Hash) (string, error) {
```
Comment (davecgh) [inline chain.go]: Same here regarding returning `*wire.BlockHeader`.

File: chain.go (hunk 1 above)
Comment (dajohi) [inline chain.go]: Agreed.

File: chain.go (hunk 2 above)
Comment (dajohi) [inline chain.go]: Updated.

**Pull request merged**